client/client_test.go

```diff
@@ -254,3 +257,42 @@ func TestUpdateRoot(t *testing.T) {
 		t.Errorf("Tree size after add Leaf: %v, want > %v", got, want)
 	}
 }
+
+func TestVerifyRootReturnsErrWhenNewRootIsNil(t *testing.T) {
```

Comment:
go style nit: Could we structure both tests as a single, table-driven test?

```go
func TestVerifyRootErrors(t *testing.T) {
  tests := []struct {
    desc string
    trusted, newRoot *trillian.SignedLogRoot
  }{
    {desc: "newRootNil", trusted: someRoot, newRoot: nil},
    {desc: "trustedNil",trusted: nil, newRoot: someRoot},
    // so on...
  }
  for _, test := range tests {
    if err := verifier.VerifyRoot(test.trusted, test.newRoot); err == nil {
      t.Errorf("%v: VerifyRoot() returned err = nil", test.desc)
    }
  }
}
```

https://github.com/golang/go/wiki/TableDrivenTests